### DISCUSSION TITLE
Using Proxy

### DISCUSSION THREAD

**Discussion by tboy1337** [masterking32/MasterHamsterKombatBot#133] [Q&A] [answered]
i have bought 50 residential static proxys but the bot does not connect to them, ive tried several times but i just cant get it to work
`"Proxy": {'http': 'http://user:password@ip:port',},`
this is what i have set, I've tried several different ways, like using double quotes instead of single, adding a trailing backslash on the end and removing the comma on the end of the proxy within the curly brackets but it just never connects also ive confirmed that the provider supports http over tcp which i think is what requests uses according to google.  Ive tested what was suggested in discord by @AmirhN1382 but that also didnt work
`        "proxies": {
"https": "https://user:password@ip:port",
"http": "http://user:password@ip:port",
},` 
Ive tested `proxies = {},` and `"proxy" = {},` this also does not work.

**Comment by camelot-rus** [comment]
`"https": "http://user:pass@10.10.1.10:3128/" `

do this and it will work)))
```

        "Proxy": {"https": "http://user:pass@10.10.1.10:3128/"},  # You can use proxies to avoid getting banned. Use {} for no proxy
        # Example of using a proxy:
        # "Proxy": {
        #   "https": "https://10.10.1.10:3128",
        #   "http": "http://user:pass@10.10.1.10:3128/"
        # },
```

**Comment by tboy1337** [comment]
Thanks for the information, I have changed it to what you recommended and now it get stuck at `Getting basic account data...` but its actually showing up on the proxy dashboard now so I guess thats some progress lol.  On the proxy dashboard it says its trying to use port 443 which is HTTPS which seems to be wrong to me.  Still not working, its driving me mad lol.

**Comment by AmirhN1382** [comment, marked as answer]
I can confirm there is no issue, I am not sure what error you're getting but here is what I did and it was just working fine:
Used residential proxy from IpRoyal,
Set both HTTP and HTTPS to `http://` proxy that i get from [IpRoyal](https://iproyal.com/?r=128279) / [WebShare](https://www.webshare.io/?referral_code=2ghmowzxmc26)
I had to disconnect my existing **VPN** connection.

![image](https://github.com/user-attachments/assets/2f3d4ae7-cc26-40b9-b180-e3ac8ad90ce1)
![image](https://github.com/user-attachments/assets/8c55ac5a-816f-40fc-9b78-820b9099b133)

**Comment by tboy1337** [comment]
I can confirm its working for me also, I think it was because I still had my VPN on.  It would probably be a good idea to add that to the readme.

**Comment by parafius** [comment]
Is the socks5 proxy supported or is there no way to connect it?

**Comment by tboy1337** [comment]
For Socks 4/5 Proxy

Run this:

Windows
```bat
pip install requests[socks]
```
Linux & Mac
```bash
pip3 install requests[socks]
```

then change in config.py:

`"Proxy": {},`
to
`"Proxy": {
    "http": "socks5://24.249.199.4:41458",
    "https": "socks5://24.249.199.4:41458"
},`
or
`"Proxy": { "http": "socks5://USERNAME:PASSWORD@24.249.199.4:41458", "https": "socks5://USERNAME:PASSWORD@24.249.199.4:41458" },`

If you are having issues then try changing ```socks5://``` to ```socks5h://```

Disconnect your VPN to test proxies

If you want to use Socks 4 change `socks5://` to `socks4://`

**Reply by parafius** [reply]
> For Socks
> 
> Run this: `pip install requests[socks]`
> 
> then change:
> 
> `{ 'http': 'socks5://24.249.199.4:41458', 'https': 'socks5://24.249.199.4:41458' }`

my proxy with password and I don't really understand, two lines in the same format to put in the proxy line?

**Reply by DineshValor** [reply]
> For Socks
> 
> Run this:`pip install requests[socks]`
> 
> then change:
> 
> `{ 'http': 'socks5://24.249.199.4:41458', 'https': 'socks5://24.249.199.4:41458' }` or `{ 'http': 'socks5://Username:Password@IPorDomain:PORT', 'https': 'socks5://Username:Password@IPorDomain:PORT' }`

![Screenshot_20240901-001118](https://github.com/user-attachments/assets/cb51aa00-73f5-4b47-8bc5-cc6b2d69055f)

Error

**Reply by AmirhN1382** [reply]
> @DineshValor 
> Error

"Max retries exceed with url:" ~ Looks like it's an Proxy ISSUE. Buy **high quality Proxies.**

**Reply by DineshValor** [reply]
> > @DineshValor
> > Error
> 
> "Max retries exceed with url:" ~ Looks like it's an Proxy ISSUE. Buy **high quality Proxies.**

Everyone have those same 10 proxy, you using shared proxies in free section.

**Reply by h-ug** [reply]
How can use proxies webshare.io need examples plz
